src/main/java/org/jvnet/hudson/test/RestartableJenkinsRule.java

```diff
@@ -188,9 +208,20 @@ public File allocate() throws Exception {
         };
 
         // run each step inside its own JenkinsRule
-        for (Statement step : steps) {
+        for (Map.Entry<Statement, Boolean> entry : steps.entrySet()) {
+            Statement step = entry.getKey();
             j = createJenkinsRule(description).with(loader);
-            j.apply(step,description).evaluate();
+            try {
+                j.apply(step, description).evaluate();
+                if (!entry.getValue()) {
+                    Assert.fail("The current JenkinsRule should have failed to start Jenkins.");
+                }
+            } catch (Exception e) {
+                if(entry.getValue()) {
+                    throw e;
+                }
+                // Failure ignored as requested
```

Comment:
Might be worth logging something to indicate we saw expected failure, so we know rule ran okay.